geomesa-core/src/main/scala/org/locationtech/geomesa/core/index/AttributeIdxStrategy.scala

```diff
@@ -161,13 +165,21 @@ trait AttributeIdxStrategy extends Strategy with Logging {
       classOf[AttributeIndexIterator]
     )
 
+    val coverage = featureType.getDescriptor(attributeName).getIndexCoverage()
+
     configureFeatureTypeName(cfg, featureType.getTypeName)
     configureFeatureEncoding(cfg, encoding)
-    configureStFilter(cfg, stFilter)
-    configureEcqlFilter(cfg, ecqlFilter.map(ECQL.toCQL))
-    configureAttributeName(cfg, attributeName)
-    configureIndexCoverage(cfg, featureType.getDescriptor(attributeName).getIndexCoverage())
     configureIndexValues(cfg, featureType)
+    configureAttributeName(cfg, attributeName)
+    configureIndexCoverage(cfg, coverage)
+    configureVersion(cfg, version)
+    if (coverage == IndexCoverage.FULL) {
+      // combine filters into one check
+      configureEcqlFilter(cfg, filterListAsAnd(Seq(stFilter ++ ecqlFilter).flatten).map(ECQL.toCQL))
```

Comment:
Covering vs non-covering indices isn't included in the query planning at this point. Priorities are:
1. if there are ID filters -> RecordStrategy
2. if there are high-cardinality attribute filters -> attribute strategy
3. if there are spatial filters -> STIDX strategy
4. Otherwise use attribute strategy if possible, else STIDX strategy
Should I add some logic for covering indices?